### PR TITLE
Fix dynamic min/max values for schedule form

### LINKED ISSUE #1939: End-time default availability in form is not compared to already selected start-time (and viceversa)
The user is able to select wrong start/end times in the availability modal form in the event-types page.
See the video for a more graphical explanation of the issue. 
[See in video](https://watch.screencastify.com/v/dxu2E7fBm5k8uUU4MoMV)

Follow-up validation is taking place well, but we should prevent reaching the step of catching the error imho and, instead, dynamically change the min/max values while the user is interacting with the form.

Another thing I noted is that the user is able to select an end time past the 24hs (up to 24:45).
I'm not sure if this is an issue or a business requirement. 

### Steps to Reproduce

1. Go to **Event Types** Page
2. Select a random event for Editing
3. Within **Advanced Features** and in the **Availability** section, click **Add Another**

### PR DESCRIPTION
**Pull request by denik1981**
## What does this PR do?

Adds a dynamic rendering for the allowed min/max values in the availability form for the Event Pages.
Currently, a user was able to select an end time prior to a start time and vice-versa. 

### RFC
This also "fixes" the allowed selection of an end time passing the 24hs. Currently, this is moving up to 24.45.
Not sure if this is a bug or a business requirement. In any case, feel free to let me know as the modification in the PR to revert back is very simple. 

Fixes #1939

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code and corrected any misspellings
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: apps/web/components/ui/form/Schedule.tsx [hunk 1]
```diff
@@ -86,8 +88,9 @@ const TimeRangeField = ({ name }: TimeRangeFieldProps) => {
               options={options}
               onFocus={() => setOptions(timeOptions())}
               onBlur={() => setOptions([])}
-              defaultValue={getOption(value)}
+              defaultValue={"getOption(value)"}
               onChange={(option) => {
+                console.log("@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@");
```
Comment (emrysal) [inline apps/web/components/ui/form/Schedule.tsx:93]: Minor bits of cleanup left, otherwise looks good - I still need to test locally

Comment (denik1981) [reply apps/web/components/ui/form/Schedule.tsx:93]: OMG, so ashamed about that. I always triple-check for that. 
Thanks for the extra pair of eyes!

Comment (denik1981) [reply apps/web/components/ui/form/Schedule.tsx]: Fixed!

**Review by emrysal — APPROVED** [review]
tested on local now; works an absolute treat and I notice no lag when adding a ton of ranges; so LGTM

**Pull request merged**